master/security/icmp-ping.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Use ICMP/ping in policy rules
+---
+
+### Big picture
+
+Use Calico network policy to allow and deny ICMP/ping messages. 
+
+### Value
+
+The **Internet Control Message Protocol (ICMP)** provides valuable network diagnostic functions, but it can also be used maliciously. Attackers can use it to learn about your network, or for DoS attacks. Using Calico network policy, you can control where ICMP is used. For example, you can:
+
+- Allow ICMP ping, but only within your cluster 
+- Allow ICMP for pods launched by operators for diagnostic purposes, but block other uses
+- Temporarily enable ICMP to diagnose a problem, then disable it after the problem is resolved
+- Deny/allow ICMP messages (ICMPv4 and/or ICMPv6)
+
+### Features
+
+This how-to guide uses the following Calico features:
+
+**GlobalNetworkPolicy** or **NetworkPolicy** with ICMPv4 and ICMPv6 and positive/negative match criteria.
+
+### Concepts
+
+#### ICMP packet type and code
+
+Calico network policy also lets you deny and allow specific parts of the ICMP packet for fine-grain control. For example, you can specify ICMP type 5, code 2. For details, see [ICMP type and code](https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol#Control_messages).
+
+### How to
+
+In this section, we introduce a global "deny all ICMP" policy to show the power of Calico network policy. Keep in mind that this policy **blocks all traffic to and from any workload in a cluster**. If your ultimate goal is to allow some traffic, be sure to have those policies ready before applying the global deny all policy.
```

Comment:
"any workload in a cluster" -> "any workload in the cluster", I think.

Reply:
Also, doesn't this make more sense to go in the "Deny all ICMP" section?

Reply:
Sure. That works too. Moved.